### ISSUE TITLE
Extract values used for saddle plot visualization

### ISSUE THREAD

**Issue by hiroyukikato911** [robinweide/GENOVA#291]
Hi,

I'm now a great fan of this tool! It's so easy going and figures are brilliant! Thank you so much!
I have some questions.

1. Is there any way to extract obs/exp value used for visualization in saddle plot?
I can find that `saddle_out` contains some calculated value, but it doesn't seem to correspond to the matrix used for visualization. 
I would appreciate it if I could extract `bin×bin txt file` containing obs/exp value that is used for visualization.

2. Or else, is there any way to visualize logFC(obs/exp) value for saddle plot just like cooltools instead of obs/exp?
I tried 
`visualise(saddle_out, contrast = 1, metric = 'lfc')`
but this doesn't seem to work.

Best regards,
Hiroyuki

**Comment by hiroyukikato911** [comment]
Hi,

Thank you so much for quick response!
I really appreciate it.

Just to make sure, the value in ‘difference’ figure is log2FC, correct?
Or is it a subtraction from control?

Best regards,

2022年3月7日月曜日 Teun van den Brand ***@***.***>:

**Comment by teunbrand** [comment]
If there are >=2 samples included in the analysis, the difference is a subtraction operation.

**Comment by hiroyukikato911** [comment]
Hi,

Thank you so much!
Actually, logFC visualization worked perfectly for my data!

Can I ask one more?
The color range seems to be continuous. To clarify, the red intensity is low in your example figure above.
Is there any way to make the Maximum value the strongest red and minimum value the strongest blue while keeping 0 to be set white?

Best regards,

**Comment by teunbrand** [comment]
Yes you can set the fill scale to an arbitrary continuous one from ggplot2. The `scale_fill_gradientn()` function allows you set colours at exact points in the (rescaled) [0, 1] range. Here is how you could calculate this:

``` r
library(GENOVA)

exp <- get_test_data("150k")

cs <- compartment_score(exp)
attr(cs, 'signed') <- TRUE # Fake signed CS
sad <- saddle(exp, cs, bins = 20)

p <- visualise(sad)
p$layers[[1]]$mapping <- ggplot2::aes(fill = log2(obsexp))

lims <- range(log2(p$data$obsexp))
values <- scales::rescale(c(lims[1], 0, lims[2]))

p + ggplot2::scale_fill_gradientn(
  colours = c("blue", "white", "red"),
  values  = values,
  limits = lims,
  name = "Log2(FC)"
)
#> Scale for 'fill' is already present. Adding another scale for 'fill', which
#> will replace the existing scale.
```

![](https://i.imgur.com/u5YICZn.png)

<sup>Created on 2022-03-08 by the [reprex package](https://reprex.tidyverse.org) (v2.0.1)</sup>

Though I must warn you that this particular type of scale goes against best practises for divergent colour scales.

**Comment by hiroyukikato911** [comment]
Hi,

Thank you so much!
The code works perfectly.
Now I can set the color bars consistent with multiple HiC experiments.

Also, I realize that calculation time seems to be faster than other tools.
What a powerful tool this is!

Best,

2022年3月8日火曜日 Teun van den Brand ***@***.***>:
